src/spec-common/injectHeadless.ts

```diff
@@ -535,12 +535,15 @@ async function runLifecycleCommand({ lifecycleHook }: ResolverParameters, contai
 			if (err && (err.code === 130 || err.signal === 2)) { // SIGINT seen on darwin as code === 130, would also make sense as signal === 2.
 				infoOutput.raw(`\r\n\x1b[1m${lifecycleHookName} interrupted.\x1b[0m\r\n\r\n`);
 			} else {
+				if (err.cmdOutput) {
```

Comment:
Should we move the error handling into `runSingleCommand` above, so it is done for each command? We might have missed that when adding the object syntax for user commands.

`runSingleCommand` uses `const printMode = name ? 'off' : 'continuous';`, so `cmdOutput` should already make it to the terminal when there are no names with the string/string-array syntax. Could you check?

Reply:
The event is thrown in runRemoteCommand function, so I have added a cmdOutput to the message to show what exactly is failed in the command.
please check the new changes committed to fix this issue.

Reply:
I suggest to keep `message` of the error short and single line. Would it instead make sense to put https://github.com/devcontainers/cli/blob/main/src/spec-common/injectHeadless.ts#L506-L517 in a try-catch block and log `cmdOutput` in the catch? Much or all of the outer catch block's code then needs to move to the inner catch: https://github.com/devcontainers/cli/blob/main/src/spec-common/injectHeadless.ts#L533-L548.

I think the 'progress' events should move out of the `runSingleCommand` function. They look like they should only be sent once for all commands of a lifecycle hook. (This is confirmed by code in the Dev Containers extension.)

Reply:
Hi @chrmarti ,
I have checked in new changes as per your suggestion. Kindly help to review it and let me know if any change is required will accordingly address the same.